src/components/Table/GridTable.tsx

```diff
@@ -404,45 +432,34 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     // Misc state to track our nested card-ification, i.e. interleaved actual rows + chrome rows
     const nestedCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style.nestedCards);
 
-    // Depth-first to filter
-    function visit(row: GridDataRow<R>, level: number): void {
-      const matches =
-        filters.length === 0 ||
-        row.pin ||
-        filters.every((filter) =>
-          columns.map((c) => applyRowFn(c, row, api)).some((maybeContent) => matchesFilter(maybeContent, filter)),
-        );
-      let isCard = false;
+    function visit(row: ParentChildrenTuple<R>, level: number): void {
```

Comment:
You could do `[row, children]: ParentChildrenTuple<R>` to clean up some of the `[0]` / `[1]` later